### PR TITLE
finished week 5 assessment

### PR DESCRIPTION
**Pull request by JAKEDAHLGREN**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,39 +14,189 @@
...
+
+const coder = (string) => {
+	return string
+		.replaceAll("a", "4")
+		.replaceAll("e", "3")
+		.replaceAll("E", "3")
+		.replaceAll("i", "1")
+		.replaceAll("o", "0")
+}
+
```
Comment (chrisyhlee) [inline code-challenges.test.js:60]: Well done!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,39 +14,189 @@
...
+// the argument is saying take the array index value, which is a string, use the string method toLowerCase and check if it includes the given letter(string), if it does return the original string value to new array, if not leave it out. 
+// return
+//
+
+// Function
+
+const fruitLetter = (array, string) => {
+	return array.filter((value) => value.toLowerCase().includes(string))
+}
+
```
Comment (chrisyhlee) [inline code-challenges.test.js:120]: Beautiful!

File: code-challenges.test.js [hunk 3]
```diff
@@ -14,39 +14,189 @@
...
+	} else if (
+		cards[0] === cards[1] &&
+		cards[0] === cards[2] &&
+		cards[3] === cards[4]
+	) {
+		return true
+	} else {
+		return false
+	}
+}
```
Comment (chrisyhlee) [inline code-challenges.test.js:184]: Nice job! Saw you had quite the party with this one haha.

File: code-challenges.test.js [hunk 4]
```diff
@@ -14,39 +14,189 @@
...
+
+// Passed
+
+// I originaly used the logic if (cards[0] === cards[1] && cards[2] === cards[4]) and else if (cards[0] === cards[2] && cards[3] === cards[4]), but it was brought to my attention that this was assuming the value for one of the cards, and it might fail on an edge case when there was a hand of unique card values.
+
+// Firt attempt
+
+// const twoThree = (array) => {
+//   array.sort
```
Comment (chrisyhlee) [inline code-challenges.test.js:193]: One thing to note is that `.sort()` needs parentheses after it. This is probably why you couldn't get this approach working, because the array was not actually being sorted!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -11,17 +20,71 @@
 # Expected output: ['tea', 'water', 'soda water']
 
 
+def letter_inside(array, letter)
+   array.select do |string| 
+   string.include?(letter)
+    end
+end
```
Comment (chrisyhlee) [inline ruby-challenges.rb:27]: Beautiful! Make sure your indentation is proper.

File: ruby-challenges.rb [hunk 6]
```diff
@@ -11,17 +20,71 @@
...
+
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
+
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
 
+def no_nest hash
+  values = hash.values.flatten
+  values.sort! 
+end
```
Comment (chrisyhlee) [inline ruby-challenges.rb:54]: Awesome! No need to make `values`, you can just slap `.sort` to the end of `.flatten`!

File: ruby-challenges.rb [hunk 7]
```diff
@@ -30,3 +93,51 @@
...
+  end
+
+  def brake braking
+  @current_speed -= braking
+  @current_speed = [0, @current_speed].max
+  # I tried this before I found the .max accessor
+  # @current_speed >= 0 
+  #   if @current_speed < 0
+  #    return 0
+  #   end
```
Comment (chrisyhlee) [inline ruby-challenges.rb:131]: Nice job researching and finding `.max`!

**Review by chrisyhlee — APPROVED** [review]
Well done, Jake! Assessment 5 down, just one more to go! This is displaying Competent level. There are definitely some refactors or different approaches you can make / take to challenge yourself, but overall you have a strong understanding of the concepts introduced in this assessment. One problem you could improve understanding on would be for the full house problem: how can you keep track of all the occurrences of any particular number in that array? Keep up the hard work!

File: code-challenges.test.js (hunk 4 above)
Comment (JAKEDAHLGREN) [reply code-challenges.test.js:193]: Man! I feel like when I transition from JavaScript to Ruby, the syntax get blurry. I will have to keep working on keeping the two separate. I will try going back through this approach to see if I can get it to work!
